### ISSUE TITLE
Not correct detect introspection in old firefox browser

### ISSUE THREAD

**Issue by Saveliev** [dfilatov/inherit#29]
Hi!
I have a problem in the old firefox browser, where not correct detect introspection.
```javascript
(function(){'_';}).toString() // "function(){}"
```

**Comment by dfilatov** [comment]
Does it cause any further problems? It's an optional optimization, everything should work anyway.

**Comment by Saveliev** [comment]
This leads to problems with inheritance of static properties, when static property represented by the class. Wrapped class loses the link to its prototype.

**Comment by dfilatov** [comment]
Could you provide a minimal example which demonstrates the problem?

**Comment by Saveliev** [comment]
```javascript
var Base = inherit({
	__constructor: function (props) {
		this.props = this.initProps(props);
	},
	initProps: function (props) {
		return props;
	},
    print: function () {
        return '';
    }
});
var Element = inherit(Base, {
    initProps: function (props) {
        return this.__base(props);
    },
    print: function () {
        return this.props.value;
    }
});
var Block = inherit(Base, {
    print: function () {
        var elem = new this.__self.Element(this.props);
        return 'value = ' + elem.print();
    }
}, {
    // link to class
    Element: Element
});
var block = new Block({value: 3});
block.print();
```

**Comment by Saveliev** [comment]
Another introspection problem occurs when the overridden `Function.prototype.toString`, this happens on pages that use `swiffy`.

**Comment by dfilatov** [comment]
fixed in 2.2.5
